src/main/java/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package dao.impl;
+
+import dao.UserDao;
+import java.util.List;
+import model.User;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class UserDaoImpl implements UserDao {
+    private final SessionFactory sessionFactory;
+
+    public UserDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public void add(User user) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(user);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert user " + user.toString(), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<User> listUsers() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("FROM User", User.class).getResultList();
```

Comment:
Would you like to use CriteriaAPI for select queries?

Reply:
Why CriteriaAPI  is more efficient?